src/live_restore/live_restore_fs.c

```diff
@@ -0,0 +1,1248 @@
+                 * the file. We're bypassing the live_restore layer so we don't try to modify the
+                 * extents in hole_list.
+                 */
+                lr_fh->destination.fh->fh_truncate(lr_fh->destination.fh, wt_session, source_size);
+
+                /*
+                 * Initialize the extent as one hole covering the entire file. We need to read
+                 * everything from source.
+                 */
+                WT_ERR(__live_restore_alloc_extent(
+                  session, 0, (size_t)source_size, NULL, &lr_fh->destination.hole_list));
+            }
+        } else
+            lr_fh->destination.complete = true;
+    }
+
+    /* Initialize the jump table. */
+    lr_fh->iface.close = __live_restore_fh_close;
+    lr_fh->iface.fh_lock = __live_restore_fh_lock;
+    lr_fh->iface.fh_read = __live_restore_fh_read;
+    lr_fh->iface.fh_size = __live_restore_fh_size;
+    lr_fh->iface.fh_sync = __live_restore_fh_sync;
+    lr_fh->iface.fh_truncate = __live_restore_fh_truncate;
+    lr_fh->iface.fh_write = __live_restore_fh_write;
+
+    /* TODO: These are unimplemented. */
+    lr_fh->iface.fh_advise = NULL;
+    lr_fh->iface.fh_sync_nowait = NULL;
+    lr_fh->iface.fh_unmap = NULL;
+    lr_fh->iface.fh_map_preload = NULL;
+    lr_fh->iface.fh_map_discard = NULL;
+    lr_fh->iface.fh_map = NULL;
+    lr_fh->iface.fh_extend = NULL;
+    lr_fh->iface.fh_extend_nolock = NULL;
+
+    *file_handlep = (WT_FILE_HANDLE *)lr_fh;
+
+    if (0) {
+err:
+        if (lr_fh != NULL)
+            __live_restore_fh_close((WT_FILE_HANDLE *)lr_fh, wt_session);
+    }
+    return (ret);
+}
+
+/*
+ * __live_restore_fs_remove --
+ *     Remove a file. We can only delete from the destination directory anyway.
+ */
+static int
+__live_restore_fs_remove(
+  WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name, uint32_t flags)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_FS *lr_fs;
+    WT_LIVE_RESTORE_FS_LAYER_TYPE layer;
+    WT_SESSION_IMPL *session;
+    char *path;
+    bool exist;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+
+    WT_UNUSED(layer);
+    exist = false;
+    path = NULL;
+
+    WT_RET_NOTFOUND_OK(__live_restore_fs_find_layer(fs, session, name, &layer, &exist));
+    if (ret == WT_NOTFOUND || !exist)
+        return (0);
+
+    /* It's possible to call remove on a file that hasn't yet been created in the destination. In
+     * these cases we only need to create the tombstone */
+    if (layer == WT_LIVE_RESTORE_FS_LAYER_DESTINATION) {
+        WT_ERR(__live_restore_fs_filename(&lr_fs->destination, session, name, &path));
+        lr_fs->os_file_system->fs_remove(lr_fs->os_file_system, wt_session, path, flags);
+    }
+
+    /* We need file tombstones here but can we be sure this is correct? */
+    __live_restore_fs_create_tombstone(fs, session, name, flags);
+    /* We don't have a file handle here so WT must have previously closed it. */
```

Comment:
Is there an open question about tombstones per the first comment? This seems fine to me.
The second comment seems outdated.

Reply:
ce9e56d4e